subcommand/webhook-cert-manager/command_test.go

```diff
@@ -369,19 +369,16 @@ func TestCertWatcher(t *testing.T) {
 	defer stopCommand(t, &cmd, exitCh)
 
 	ctx := context.Background()
-	// This first sleep is required to ensure the command had completed validating that the MWC does exist in the cluster
-	// It ensures the MWC is absent when the server actually attempts to update it with a bundle.
-	time.Sleep(1 * time.Millisecond)
+	// This sleep ensures the webhook has been updated and the reconcile has been successful.
+	time.Sleep(100 * time.Millisecond)
 	err = k8s.AdmissionregistrationV1beta1().MutatingWebhookConfigurations().Delete(ctx, webhookName, metav1.DeleteOptions{})
 	require.NoError(t, err)
 
-	// This sleep ensures the cluster attempts to try updating the MWC every second and fails before we re-create it in the cluster.
-	time.Sleep(100 * time.Millisecond)
 	_, err = k8s.AdmissionregistrationV1beta1().MutatingWebhookConfigurations().Create(ctx, webhook, metav1.CreateOptions{})
 	require.NoError(t, err)
 
-	// If this test passes, it implies that the system has recovered before a new certificate is available
-	// or its default retry timer of 30 minutes has forced the reconcile.
+	// If this test passes, it implies that the system has recovered from the MWC
+	// getting deleted and restored withing a reasonable time window
 	timer := &retry.Timer{Timeout: 10 * time.Second, Wait: 500 * time.Millisecond}
 	retry.RunWith(timer, t, func(r *retry.R) {
 		webhookConfig1, err := k8s.AdmissionregistrationV1beta1().MutatingWebhookConfigurations().Get(ctx, webhookName, metav1.GetOptions{})
```

Comment:
```
		require.NotEqual(r, webhookConfig1.Webhooks[0].ClientConfig.CABundle, "")
		// This verifies that the MWC is populated with a valid certificate which has a
		// a len of 800 chars. But the length is not deterministic because it is encoded.
		require.True(r, len(webhookConfig1.Webhooks[0].ClientConfig.CABundle) > 800)
```

Can be
```
		// This verifies that the MWC is populated with a valid certificate which has a
		// a len of 800 chars. But the length is not deterministic because it is encoded.
		require.Len(r, 800, webhookConfig1.Webhooks[0].ClientConfig.CABundle)
```

Reply:
The length of the certificate is non deterministic. I can drop the `""` check but the base64 encoding of the ca bundle varies a little bit.

Reply:
Ahh gothca, but yeah we ca just assert len > 800